### ISSUE TITLE
Team management

### ISSUE THREAD

**Issue by crypterexe** [moloch--/RootTheBox#162]
Hello RootTheBox team, 

I have some problem with platform, my issue is anyone can be added to any team !

**Comment by jeffg2k** [comment]
The user has the ability to set their team.  Are you asking for an option that only the Administrator can set a player's team?  There is an option to disable the ability for users to create teams and making it so only the administrator can create teams, but the user is still allowed to select among those teams when they register.

**Comment by mohammadHmedat** [comment]
Hello RootTheBox Team,

 @crypterexe  is my partner on the ctf project, at the first i would to thank you for the awesome effort on the platform and for the fast reply, actually our problem that any new user can add himself to any created team by other users, "ex: suppose i create a team name called TestTeam for my team members, any new user can add himself to my team but i want just my team in the group!!" so if there any way to solve this issue like: determine team leader (who created the team), then if any new user add himself to his team he must receive new notification to accept this user in his team or not !!

Another problem is if a new user create a team with the same name of exist team he will be added to the created team !!

We will be so thank for you for help

**Comment by jeffg2k** [comment]
@moloch-- any thoughts?  Might be easy to add a team password upon creation, so if someone wants to join the team, they have to supply the correct password.  This might add a bit of a UI overload with three possible passwords (user, team, bank), but would be easier than creating a message / authorization system.

**Comment by moloch--** [comment]
Team password is probably the easiest approach, I think the other option is to give the user that created the team a special permission to control membership to the group/team.

**Comment by F13** [comment]
An alternative approach might be to require an existing team member to authorize the addition of any new members.

**Comment by jeffg2k** [comment]
@mohammadHmedat @crypterexe depending on how soon your game is, you could have the admin completely control team association by turning off public team creation in the config and then removing the "Select a Team" code from templates/public/registration.html

I'll try to work on it, but it's a bit more involved than editing a couple files and creating a db entry.  It's going to take a little bit of time.

**Comment by jeffg2k** [comment]
Disregard that last message regarding removing Select a Team.  I tested it and it wouldn't work.  I'll try to work on the password approach.  I'm concerned that the team approval approach may put someone in limbo pending approval / ignored before they could play the game.

**Comment by mohammadHmedat** [comment]
Great we waiting for your work soon

Thanks.

**Comment by moloch--** [comment]
Perhaps a mix, just call it like a "Team Code" (avoid calling it a 'password') or something --randomly generated ~64bit identifier, and anyone on the team can see it and thus provide it to anyone who wants to join the team.

**Comment by jeffg2k** [comment]
Ok, I pushed some code for this.  
This update adds two new features.

1. It adds a Team Code (currently 128bit but we can shorten it to 64bit - that was my error when writing it as I just used uuid4().hex).  The code is then displayed on the player's home page.
2. It adds an option to display an MVP list on the Scoreboard when in Team Mode.  Gave player points for flags (and deductions for penalties).  Only displays players that have earned the team points.

There are a couple DB updates if you're updating an existing RTB.
```
ALTER TABLE `rootthebox`.`team` 
ADD COLUMN `code` VARCHAR(32) NULL DEFAULT NULL AFTER `money`;
 ALTER TABLE `rootthebox`.`user` 
ADD COLUMN `money` INT(11) NOT NULL DEFAULT 0 AFTER `algorithm`;
```
